### PR TITLE
feature/ bug fixes

### LINKED ISSUE #91: Viewing a Guest's Details on tapping on their name
Tapping on the guest's name in their profile should show their details i.e. name, email, ticket details

### LINKED ISSUE #106: When a guestlist is opened on the iOS app and you click on a contact you receive a screen with a pink square block
Steps to reproduce:

- When a guestlist is opened on the iOS app and you click on a guest you receive a screen with a pink square block with no other information showing, only the below screen pops up.

### LINKED ISSUE #108: When starting any API request first check if the access_token has already expired
image

Before starting any request check if the expiry time of the `access_token.exp > now() + 60 seconds` if the local value of the JWT is expiry within the next 60 seconds, fetch a new access token from the server and continue with the request.

If the refresh API call returns a 401, log the user out and return to the login page.

One example of this being required is https://github.com/big-neon/bn-ios/blob/10a4322b166f1acc18a44bd99a5a95c53f6f49eb/Big%20Neon%20Core/Services/DatabaseService/DatabaseService.Scan.swift#L64

[ ] Check JWT expiry on every request.
[ ] Request new access token on near expired tokens
[ ] Logout user on 401 to refresh URL.

### LINKED ISSUE #112: On login change the email input to an email type

### LINKED ISSUE #113: On logout wipe all local cache.
When a user logs out and switches schemes from Beta to Development, the app crashes on 
this line after logging back into the app. 

https://github.com/big-neon/bn-ios/blob/10a4322b166f1acc18a44bd99a5a95c53f6f49eb/Big%20Neon/Offline%20Config/EventsFetcher.swift#L50

### LINKED ISSUE #114: Swipe to checkin on the guest list enhancement
Swipe to  75% of the way on a valid ticket and the redemption process begins, the bar then completely slides out and back in with a spinner to show the redemption process is underway.

### PR DESCRIPTION
**Pull request by mhlangagc**
### Closes Issues:
- closes #91 
- closes #114 
- closes #112 
- closes #113 
- closes #106
- closes #108  

### Description:
- Shows a guest in a separate view
- Fixes the scanning issues 

### Pods Installed
 * No change

### CODE REVIEW

File: Big Neon/Screens/Explore/GuestListViewController/GuestListViewController.TableView.swift [hunk 1]
```diff
@@ -87,6 +86,7 @@ extension GuestListViewController: SwipeActionTransitioning {
     
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
+        //  self.showGuest() - To be added later. A way to select and show the guest list.
     }
```
[the remark this answers is not in the document]
Comment (mhlangagc) [reply Big Neon/Screens/Explore/GuestListViewController/GuestListViewController.TableView.swift]: I added this part for later. Will need it for a later sprint

File: Big Neon/Screens/Explore/GuestListViewController/GuestListViewController.TableView.swift [hunk 2]
```diff
@@ -71,10 +70,10 @@ extension GuestListViewController: SwipeActionTransitioning {
         guestCell.ticketTypeNameLabel.text = price.dollarString + " | " + guestValues!.ticketType + " | " + ticketID
         
         if guestValues!.status == TicketStatus.purchased.rawValue {
-            guestCell.ticketStateView.tagLabel.text = "PURCHASED"
+            guestCell.ticketStateView.setTitle("PURCHASED", for: UIControl.State.normal)
             guestCell.ticketStateView.backgroundColor = UIColor.brandGreen
```
[the remark this answers is not in the document]
Comment (mhlangagc) [reply Big Neon/Screens/Explore/GuestListViewController/GuestListViewController.TableView.swift]: Will do in a new sprint. :)

File: Big Neon/Screens/Explore/DoorPersonViewController/DoorPersonViewModel.swift [hunk 3]
```diff
@@ -47,66 +50,70 @@ final class DoorPersonViewModel {
...
-            })
+            switch tokenResult {
+            case .noAccessToken?:
+               print("No Access Token Found")
+               completion(false)
+            case .tokenExpired?:
+                print("Token has expired")
+                completion(false)
+            default:
+                BusinessService.shared.database.fetchCheckins { [weak self] (error, events) in
```
[the remark this answers is not in the document]
Comment (mhlangagc) [reply Big Neon/Screens/Explore/DoorPersonViewController/DoorPersonViewModel.swift]: Good Idea :)

File: Big Neon Core/Services/TokenService/TokenService.swift [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                    return
+                }
+                completion(.tokenRefreshed, nil)
+            }
+            return
+        }
+        
+        do {
+            let jwt = try decode(jwt: accessToken)
+            let _ = jwt.expired
```
Comment (Krakaw) [inline Big Neon Core/Services/TokenService/TokenService.swift]:
```suggestion
            let _ = jwt.exp
```

File: Big Neon Core/Services/TokenService/TokenService.swift [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+            guard let tokenExpiryDate = jwt.expiresAt else {
```
Comment (Krakaw) [inline Big Neon Core/Services/TokenService/TokenService.swift]:
```suggestion
            guard let tokenExpiryDate = jwt.exp else {
```

Comment (mhlangagc) [reply Big Neon Core/Services/TokenService/TokenService.swift]: The `DecodedJWT` library I am using takes care of this and abstracts `exp` into `expiresAt` for readability.

**Pull request merged**